### PR TITLE
#1524 added "if-equal" header to define whether to "skip" and update of an equal value

### PR DESCRIPTION
**Pull request by thjaeckle**
* default is "update" (which is the current behavior), always overwriting the value, even if it is equal to the one before
* return a "*PreconditionNotModifiedException" (HTTP 304) when "skip" is provided and value is equal
* work on logic is still in progress

Resolves: #1524

Open:

- [x] unit tests
- [x] documentation

### CODE REVIEW

**Comment by thjaeckle** [comment]
@alstanchev @kalinkostashki I would appreciate a review on this - turned out to be not too complicated to do :)

**Review by kalinkostashki — COMMENTED** [review]
@thjaeckle IfEqualsOption is maybe a bit of strange name for a DittoHeader ? 
Shouldn't it be IfEquals only ? Or maybe IfEqualsHeader?
I don't have any hard feelings about IfEqualsOption -> just find it maybe a bit weird :)

File: documentation/src/main/resources/pages/ditto/httpapi-concepts.md [hunk 1]
```diff
@@ -621,6 +621,12 @@ The following request headers can be used to issue a conditional request:
               information and the current entity-tag of the resource as `ETag` header
             * for read requests, status `304 (Not Modified)` without response body, with the current entity-tag of the
               resource as `ETag` header
+* `if-equal`:
+   * The `if-equal` header can take the values `'update'` (which is the default if omitted) or `'skip'`
+   * Write the resource only
+      * in case `if-equal: 'update'` is defined - even if the entity will be equal than before the update.
```
[the remark this answers is not in the document]
Comment (thjaeckle) [reply documentation/src/main/resources/pages/ditto/httpapi-concepts.md]: Thx, makes sense 👍

File: internal/utils/conditional-headers/src/main/java/org/eclipse/ditto/internal/utils/headers/conditional/ConditionalHeadersValidator.java [hunk 2]
```diff
@@ -171,6 +208,12 @@ private DittoRuntimeException buildNotModifiedException(final PreconditionHeader
                 .build();
     }
 
+    private DittoRuntimeException buildNotModifiedForEqualityException() {
```
[the remark this answers is not in the document]
Comment (thjaeckle) [reply internal/utils/conditional-headers/src/main/java/org/eclipse/ditto/internal/utils/headers/conditional/ConditionalHeadersValidator.java]: Right, that seems to be a leftover from a "WIP" version. Removed it ..

**Review by kalinkostashki — APPROVED** [review]
LGTM :)

**Pull request merged**